### PR TITLE
Add image component

### PR DESCRIPTION
**Pull request by pandermatt**
Added ru-img (Image Component and Image Service) and a read me how to use

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,30 @@ This is a component for the renuo upload (https://github.com/renuo/renuo-upload)
...
+<ru-img [src]="picture"></ru-img>
+```
+
+#### Advanced usage
+```angular2html
+<ru-img [src]="picture"  
+    [dimensions]="'400x400'" 
+    [quality]="10" 
+    [roundedCorners]="60" [background]="'10,10,255'"
+    [watermark]="'https://www.renuo.ch/images/logo.png'">
```
Comment (lukaselmer) [inline README.md]: i do not think that there is a need for the `[]`

File: README.md [hunk 2]
```diff
@@ -13,7 +13,30 @@ This is a component for the renuo upload (https://github.com/renuo/renuo-upload)
 
 ## How to use?
 
-Coming Soon
+### Single Image
+#### Simple usage
+```angular2html
+<ru-img [src]="picture"></ru-img>
```
Comment (lukaselmer) [inline README.md]: It would be nice to have a real example with a real image which works

File: src/app/demo/demo.component.ts [hunk 3]
```diff
@@ -8,4 +8,6 @@ import { I18n } from '../i18n/i18n';
 })
 export class DemoComponent {
   i18n = I18n;
+  picture: string = '//renuo-upload-cdn-master.renuoapp.ch/o/lawoon-develop/ch6d/060f/21fd/a6d7/9318/441c/8483/e11e/5acf/' +
+    'randomuser-woman-1.jpg';
```
Comment (lukaselmer) [inline src/app/demo/demo.component.ts]: Please be careful with copyright

File: src/app/image/image.component.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+@import '../scss/variables.style';
```
Comment (lukaselmer) [inline src/app/image/image.component.scss]: I would delete this since no variables are needed here

File: src/app/image/image.component.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+img {
+  width: auto;
```
Comment (lukaselmer) [inline src/app/image/image.component.scss]: Why is this needed?

File: src/app/image/image.component.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  @Input() roundedCorners: number;
+  @Input() background: string;
+  @Input() watermark: string;
+  @Input() src: string;
+  displayedImage: string;
+  i18n = I18n;
+
+  constructor(private imageService: ImageService) {}
+
+  ngOnInit() {
```
Comment (lukaselmer) [inline src/app/image/image.component.ts:21]: This seems complicated. Why not create a method which returns the right url instead?

File: src/app/image/image.component.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import { I18n } from '../i18n/i18n';
+import { ImageService } from '../services/image/image.service';
+
+@Component({
+  selector: 'ru-img',
+  styleUrls: ['image.component.scss'],
+  templateUrl: 'image.component.html'
+})
+export class ImageComponent implements OnInit {
+  @Input() dimensions: string;
```
Comment (lukaselmer) [inline src/app/image/image.component.ts]: Would the type not be `string?`?

File: src/app/image/image.component.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { I18n } from '../i18n/i18n';
+import { ImageService } from '../services/image/image.service';
+
+@Component({
+  selector: 'ru-img',
+  styleUrls: ['image.component.scss'],
+  templateUrl: 'image.component.html'
+})
+export class ImageComponent implements OnInit {
```
Comment (lukaselmer) [inline src/app/image/image.component.ts]: I prefer the selector and the component name to be the same (except the prefix)

File: src/app/services/image/image.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  describe('when there are all filters and a dimenson', () => {
+    it('returns the modified image, with watermark filter', () => {
+      const filters = {quality: 20, roundedCorners: 50, background: '20,200,10', watermark: 'https://www.renuo.ch/images/logo.png'};
+      const modifiedImage = service.modifyImage(testImageUrl, '400x500', filters);
+      const expectedFilters = 'filters:quality(20):watermark(https://www.renuo.ch/images/logo.png,-10,-10,50):round_corner(50,20,200,10)';
+      expect(modifiedImage).toEqual(`https://${host}/t/400x500/${expectedFilters}/u${imagePath}`);
+    });
+  });
```
Comment (lukaselmer) [inline src/app/services/image/image.service.spec.ts:44]: :+1:

File: src/app/services/image/image.service.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    const filtersUrl = this.buildFilters(filters);
+
+    if (dimensionsUrl || filtersUrl) {
+      return this.thumborImage(url, dimensionsUrl + filtersUrl);
+    } else {
+      return url;
+    }
+  }
+
+  thumborImage(url: string, imageOptions: string) {
```
Comment (lukaselmer) [inline src/app/services/image/image.service.ts]: Does this need to be public?

**Review by lukaselmer — APPROVED** [review]
There are some small thing, but otherwise it lgtm :+1:

File: README.md (hunk 2 above)
Comment (pandermatt) [reply README.md]: The Example will be linked to the demo page

**Review by lukaselmer — APPROVED** [review]
There are still some small things open, but it is your decision if you would like to merge it like this :)
